wicket-jquery-ui-parent/wicket-jquery-ui-calendar/src/main/java/com/googlecode/wicket/jquery/ui/calendar6/CalendarEvent.java

```diff
@@ -0,0 +1,310 @@
+	}
+
+	public void setSource(String source)
+	{
+		this.source = source;
+	}
+
+	public Boolean isAllDay()
+	{
+		return this.allDay;
+	}
+
+	public void setAllDay(Boolean allDay)
+	{
+		this.allDay = allDay;
+	}
+
+	public LocalDateTime getStart()
+	{
+		return this.start;
+	}
+
+	public void setStart(LocalDateTime start)
+	{
+		this.start = start;
+	}
+
+	public LocalDateTime getEnd()
+	{
+		return this.end;
+	}
+
+	public void setEnd(LocalDateTime end)
+	{
+		this.end = end;
+	}
+
+	/**
+	 * Overrides the master editable option for this single event.
+	 *
+	 * @return true or false
+	 */
+	public Boolean isEditable()
+	{
+		return this.editable;
+	}
+
+	public void setEditable(Boolean editable)
+	{
+		this.editable = editable;
+	}
+
+	// styling //
+	public String getClassName()
+	{
+		return this.className;
+	}
+
+	public void setClassName(String className)
+	{
+		this.className = className;
+	}
+
+	public String getColor()
+	{
+		return this.color;
+	}
+
+	public void setColor(String color)
+	{
+		this.color = color;
+	}
+
+	public void setColor(String backgroundColor, String borderColor)
+	{
+		this.backgroundColor = backgroundColor;
+		this.borderColor = borderColor;
+	}
+
+	public String getBackgroundColor()
+	{
+		return this.backgroundColor;
+	}
+
+	public void setBackgroundColor(String backgroundColor)
+	{
+		this.backgroundColor = backgroundColor;
+	}
+
+	public String getBorderColor()
+	{
+		return borderColor;
+	}
+
+	public void setBorderColor(String borderColor)
+	{
+		this.borderColor = borderColor;
+	}
+
+	public String getTextColor()
+	{
+		return this.textColor;
+	}
+
+	public void setTextColor(String textColor)
+	{
+		this.textColor = textColor;
+	}
+
+	/**
+	 * Create an {@link Options} JSON-object that will embed the (default) properties of this {@link CalendarEvent}
+	 *
+	 * @return a new {@code Options} object
+	 */
+	protected Options createOptions()
+	{
+		Options options = new Options();
+
+		options.set("id", this.id);
+
+		if (this.title != null)
+		{
+			options.set("title", Options.asString(this.title));
+		}
+
+		if (this.start != null)
+		{
+			options.set("start", Options.asString(this.start));
+		}
+
+		if (this.end != null)
+		{
+			options.set("end", Options.asString(this.end));
+		}
+
+		if (this.url != null)
+		{
+			options.set("url", Options.asString(this.url));
+		}
+
+		if (this.source != null)
+		{
+			options.set("source", Options.asString(this.source));
+		}
+
+		if (this.allDay != null)
+		{
+			options.set("allDay", this.allDay);
+		}
+
+		if (this.editable != null)
+		{
+			options.set("editable", editable);
+		}
+
+		// styling //
+		if (this.className != null)
+		{
+			options.set("className", Options.asString(this.className));
+		}
+
+		if (this.color != null)
+		{
+			options.set("color", Options.asString(this.color));
+		}
+
+		if (this.backgroundColor != null)
+		{
+			options.set("backgroundColor", Options.asString(this.backgroundColor));
+		}
+
+		if (this.borderColor != null)

```

Comment:
The code could probably be much reduced by having an alternative option.set method (maybe setUnlessNull).